### ISSUE TITLE
Extended calendar menu hover area

### ISSUE THREAD

**Issue by bordam** [CleoMenezesJr/gnome-shell-extension-weather-oclock#1]
## Distro Name & Version
Arch Linux Rolling

## GNOME Shell Version

43

## Other Installed Extensions

None

## Expected Behavior

I should only be able to open the calendar/notification menu when I hover on the date and time

## Current Behavior

The area where I can open the menu is extended when extension is active

## Detailed Description

[Video del 2022-11-04 15-00-02.webm](https://user-images.githubusercontent.com/95233098/199990778-a2d3267a-705a-466c-91af-471dc1a9a449.webm)

**Comment by somepaulo** [comment]
I think that with this behaviour the weather info should be displayed inside the date/time pill, making it truly part of the clock.
Alternatively, it should have its own pill and open the Weather app on click.
And a toggle to choose which side of the clock/date to display the weather would be fantastic.

**Comment by Anifyuli** [comment]
> I think that with this behaviour the weather info should be displayed inside the date/time pill, making it truly part of the clock. Alternatively, it should have its own pill and open the Weather app on click. And a toggle to choose which side of the clock/date to display the weather would be fantastic.

I agree with you, Mister. So this extension can get better integration to shell

**Comment by CleoMenezesJr** [comment]
> 
@somepaulo, the idea is to some day put it inside the same pill as date/time, I just didn't have time to work on it.
Btw, I don't want it to be in a separate pill because of the idea I already mentioned and I don't want it to be configurable since I want to keep it simple.

**Comment by somepaulo** [comment]
> I don't want it to be in a separate pill because of the idea I already mentioned

Fair enough. My idea and purpose are different. So I forked your extension to do what I wanted - show a separate pill to the right of the clock, and open GNOME Weather on click.
https://github.com/somepaulo/gnome-shell-extension-weather-or-not

**Comment by CleoMenezesJr** [comment]
> > I don't want it to be in a separate pill because of the idea I already mentioned
> 
> Fair enough. My idea and purpose are different. So I forked your extension to do what I wanted - show a separate pill to the right of the clock, and open GNOME Weather on click. https://github.com/somepaulo/gnome-shell-extension-weather-or-not

Good! I hope you have a lot of success.

**Comment by Alkotronikk** [comment]
Would it be possible to add a setting to disable that extra space to the right? I know it's not an intended usecase, but if I have the clock moved to the right of the screen then the extra space does not make sense for me.

![image](https://user-images.githubusercontent.com/8993765/212040201-841fd042-0210-4991-9782-909a5e823559.png)

**Comment by CleoMenezesJr** [comment]
> Would it be possible to add a setting to disable that extra space to the right? I know it's not an intended usecase, but if I have the clock moved to the right of the screen then the extra space does not make sense for me.
> 
> ![image](https://user-images.githubusercontent.com/8993765/212040201-841fd042-0210-4991-9782-909a5e823559.png)

It's unexpected behavior, but I'm taking the time for the prediction inside the pill, so it will avoid any of these problems.

**Comment by somepaulo** [comment]
> Good! I hope you have a lot of success.

Thank you! Igualmente!

> I'm taking the time to implement this inside the pill

While modifying your code to my purpose, I've read quite a lot of tutorials, explanations and code examples, and I came across a piece of code that gave me an idea for how to put your extension into the clock pill.

You might want to use a topBox to combine your icon and label together (otherwise only the icon might get displayed).
```js
let topBox = new St.BoxLayout({
  style_class: 'panel-status-menu-box'
});
topBox.add_child(this._icon);
topBox.add_child(this._label);
this.add_child(topBox);
```

And then, the solution itself would be to get the label from the clock into a variable and redefine the label with the output of the `PanelWeather` class plus the original label from the variable.

**Comment by somepaulo** [comment]
I came across [this extension](https://github.com/stuartlangridge/gnome-shell-clock-override) that overrides stuff in the clock. You might want to look into its code to see how to get the weather inside the clock pill.

**Comment by CleoMenezesJr** [comment]
> I came across [this extension](https://github.com/stuartlangridge/gnome-shell-clock-override) that overrides stuff in the clock. You might want to look into its code to see how to get the weather inside the clock pill.

Thank you for your help. I'm taking a similar path. I'm creating a BoxLayout that contains panelWeather and the object that contains the datetime. Still need to make the datetime dynamic, but overall the idea seems to work

**Comment by CleoMenezesJr** [comment]
[Screencast from 2023-01-16 02-11-09.webm](https://user-images.githubusercontent.com/54215258/212603213-37ada9a2-ec0b-4543-8ea2-84443666cb93.webm)

**Comment by Anifyuli** [comment]
> Screencast.from.2023-01-16.02-11-09.webm

I like it, Mister. I wish you can release it soon as possible. I think that's cool from previous looks

**Comment by CleoMenezesJr** [comment]
> > Screencast.from.2023-01-16.02-11-09.webm
> 
> I like it, Mister. I wish you can release it soon as possible. I think that's cool from previous looks


I need a few hours of sleep during the week, and maybe I'll be ready the next week. hahaha
